spec/requests/admin/site_settings_controller_spec.rb

```diff
@@ -20,6 +20,25 @@
 
         expect(locale.length).to eq(1)
       end
+
+      describe "the filter_names param" do
+        it "only returns settings that are specified in the filter_names param" do
+          get "/admin/site_settings.json",
+              params: {
+                filter_names: %w[title site_description notification_email],
+              }
+
+          expect(response.status).to eq(200)
+
+          json = response.parsed_body
+          expect(json["site_settings"].size).to eq(3)
+          expect(json["site_settings"].map { |s| s["setting"] }).to contain_exactly(
```

Comment:
Just double checking this returns all the other setting details right so the value, validation (e.g. min/max) and so on?

Reply:
Yup, e.g.:

```
[2] pry(main)> SiteSetting.all_settings.find { |ss| ss[:setting] == :suggested_topics }
  ColorScheme Load (0.7ms)  SELECT "color_schemes".* FROM "color_schemes"
=> {:setting=>:suggested_topics,
 :description=>"Number of suggested topics shown at the bottom of a topic.",
 :keywords=>[""],
 :default=>"5",
 :value=>"5",
 :category=>"basic",
 :preview=>nil,
 :secret=>false,
 :placeholder=>nil,
 :mandatory_values=>nil,
 :type=>"integer",
 :max=>2000}
```

`SiteSetting.all_settings` is what the controller action calls to retrieve the settings.